### ISSUE TITLE
TableSchema.columns as OrderedDictionary

### ISSUE THREAD

**Issue by chukarsten** [alteryx/woodwork#1039]
- As an EvalML developer, I would like Woodwork TableSchema to preserve column order of the underlying DataFrame so that DataFrame behavior is more closely emulated.

It would be nice if the following test would pass.  It would allow us to remove a function from our Woodwork compatibility library.  As a caveat, this test may or may not pass as the order of the resulting columns is not guaranteed to be consistent.

#### Code Example

```python
import pytest
@pytest.mark.parametrize(
    "ww_type,pandas_type",
    [("Integer", "int"),
     ("Double", "float"),
     ("Datetime", "datetime"),
     ("NaturalLanguage", "string"),
     ("Categorical", "category"),
     ("Boolean", "boolean")]
)
def test_ww_column_order(ww_type, pandas_type):

    df = ww.demo.load_retail(nrows=100, init_woodwork=False)
    df.ww.init(name="retail")
    ww_schema_column_order = [x for x in df.ww.select([ww_type], return_schema=True).columns.keys()]
    pandas_column_order = [x for x in df.select_dtypes(include=pandas_type).columns]
    assert ww_schema_column_order== pandas_column_order

```

**Comment by chukarsten** [comment]
@freddyaboulton would this allow us to revert that function you're introducing?

**Comment by freddyaboulton** [comment]
Yea the issue we saw is that `X.ww.select(...)` returns the columns in the order they appear in X. We want to use `X.ww.select(..., return_schema=True).columns` but the columns are not (guaranteed) to be in the order they appear in X, which can cause problems.

Using an OrderedDictionary is one possible solution but the ww team may have alternative solutions!

**Comment by thehomebrewnerd** [comment]
@chukarsten @freddyaboulton This behavior is due to code in the `TableSchema._filter_cols` method. In that method we had used a set to collect matching columns since it is possible a column could match multiple filters (for example both the `Integer` logical type and `numeric` standard tag, if both were used). To make sure we only included the column once, a set was used, but this caused our column order guarantee to be lost.

This method can be updated to store the matching columns in a list instead, which will preserve order, along with some additional logic to make sure that duplicate columns aren't added to the list. This will likely cause a small performance hit, but I'm assuming that maintaining column order is more important?
